src/Qir/Runtime/stdlib/src/arrays.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#![deny(clippy::all, clippy::pedantic)]
+
+use crate::update_counts;
+use std::{rc::Rc, usize};
+
+#[derive(Debug, Clone)]
+pub struct QirArray {
+    pub(crate) elem_size: usize,
+    pub(crate) data: Vec<u8>,
+}
+
+/// # Panics
+///
+/// This function panics if the passed in sizes do not fit into the usize type for the
+/// current platform.
+#[no_mangle]
+pub extern "C" fn __quantum__rt__array_create_1d(elem_size: u32, size: u64) -> *const QirArray {
```

Comment:
FYI: About the second param `size: u64`: I personally prefer to call the number of elements in an array (or number of chars in a string) with the word "length" or "itemCount" or "count", and I prefer to use the word "size" to designate the _size in bytes_.
Some more details are in [Distinguish Between Size and Length](https://github.com/kuzminrobin/code_review_notes/blob/master/cpp_design_bookmarks.md#distinguish-between-size-and-length).

Consider renaming the second param:
```suggestion
pub extern "C" fn __quantum__rt__array_create_1d(elem_size: u32, itemCount: u64) -> *const QirArray {
```